src/views/admin/components/Application/ApplicationSection/ApplicationSection.tsx

```diff
@@ -0,0 +1,72 @@
+import { ReactElement } from "react";
+import { Col, Row } from "react-bootstrap";
+import { Applicant } from "views/admin/applications/AdminHome";
+
+interface ApplicationSectionProps {
+	title: string;
+	data: Applicant;
+	propsToShow: Array<string>;
+}
+
+function generateElement(data: Applicant, prop: string): ReactElement {
+	const value = data[prop as keyof Applicant];
+	if (!value) {
+		return <p>Not Provided</p>;
+	}
+
+	switch (typeof value) {
+		case "boolean":
+			return <p>{value ? "Yes" : "No"}</p>;
+		case "string":
+			if (value.startsWith("http")) {
+				return (
+					<p>
+						<a href={value} target="_blank" rel="noopener noreferrer">
+							{value}
+						</a>
+					</p>
+				);
+			}
+			return <p>{value}</p>;
+		case "object":
+			return (
+				<ul>
+					{value.map((v, i) => (
+						<li key={i}>{v}</li>
+					))}
+				</ul>
+			);
+		default:
+			return <p />;
+	}
+}
+
+function ApplicationSection({
+	title,
+	data,
+	propsToShow,
+}: ApplicationSectionProps) {
+	return (
+		<div className="mb-5">
+			<h2>{title}</h2>
+			<Row>
+				{propsToShow.map((prop, index) => {
+					const element = generateElement(data, prop);
```

Comment:
Note: Calling a function component instead of rendering it as an element may lead to unexpected behavior.

Reply:
Resolved in e047f80.